core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -724,8 +977,7 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
   // check epoch + 1. Epoch bumps from PrepareEpochFence state are handled separately, so this method should not be used to check that case.
   // Returns true if the transaction state epoch is the specified producer epoch + 1 and epoch bump on every transaction is expected.
   private def endTxnEpochBumped(txnMetadata: TransactionMetadata, producerEpoch: Short): Boolean = {
-    !txnMetadata.pendingState.contains(PrepareEpochFence) && txnMetadata.clientTransactionVersion.supportsEpochBump() &&
-      txnMetadata.producerEpoch == producerEpoch + 1
+    !txnMetadata.pendingState.contains(PrepareEpochFence) && txnMetadata.producerEpoch == producerEpoch + 1
```

Comment:
I'm not sure we can check `txnMetadata.pendingState.contains(PrepareEpochFence)` -- when we call this function from the `sendTxnMarkersCallback` the pendingState is already something else.   We need to pass the `isEpochFence` as an argument.

Reply:
Ummm, from the method usage perspective, inside of the sendTxnMarkersCallback, the epoch should be bumped no matter if the epoch is previously fenced. The epoch should be bumped anyway.

Reply:
Removed the endTxnEpochBumped